crates/librqbit_core/src/magnet.rs

```diff
@@ -2,11 +2,49 @@ use std::str::FromStr;
 
 use anyhow::Context;
 
-use crate::id20::Id20;
+use crate::hash_id::{Id20, Id32};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum InfoHash {
+    V1(Id20),
```

Comment:
I agree that a single info-hash can be an enum of either V1 or V2.
However, if you imagine a magnet link, can a magnet link exist which provides both hashes? If so, the current Magnet struct doesn't account for that. Again, if that is true, maybe change Magnet to be ```struct Magnet {id20: Option<Id20>, id32: Option<Id32>}``` and move the "as_id20" like method into Magnet rather than InfoHash

Reply:
>  can a magnet link exist which provides both hashes?

Very much yes, that was a silly oversight on my part. They're called hybrid magnet links, they look like this:
```
magnet:?xt=urn:btih:631a31dd0a46257d5078c0dee4e66e26f73e42ac&xt=urn:btmh:1220d8dd32ac93357c368556af3ac1d95c9d76bd0dff6fa9833ecdac3d53134efabb&dn=bittorrent-v1-v2-hybrid-test
```

The only issue with `struct Magnet {id20: Option<Id20>, id32: Option<Id32>}` is that it would be possible to represent a magnet link without an infohash if both are `None`. Not a huge deal but something to keep in mind

Reply:
That's fine, as long as magnet construction / parsing fails when there's 0 hashes, and if the fields are private, it would be impossible to construct a broken magnet.